backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/invoice/impl/JsonInvoiceService.java

```diff
@@ -669,18 +669,10 @@ private DocTypeId getDocTypeId(
 			@NonNull final JsonDocTypeInfo invoiceDocType,
 			@NonNull final ClientAndOrgId clientAndOrgId)
 	{
-		final DocTypeId docTypeId = docTypeService.getDocTypeId(
+		return docTypeService.getDocTypeId(
 				DocBaseType.ofCode(invoiceDocType.getDocBaseType()),
 				invoiceDocType.getDocSubType(),
 				clientAndOrgId.getOrgId());
-
-		return Optional.ofNullable(docTypeId)
-				.orElseThrow(() -> new AdempiereException("No DocumentType found!")
-						.appendParametersToMessage()
-						.setParameter("ClientId", clientAndOrgId.getClientId())
-						.setParameter("OrgId", clientAndOrgId.getOrgId())
-						.setParameter("DocBaseType", invoiceDocType.getDocBaseType())
-						.setParameter("DocSubType", invoiceDocType.getDocSubType()));
```

Comment:
Note `docTypeService.getDocTypeId` is `@NonNull` or throws DocumentNotFoundException